ncm-network/src/main/pan/components/network/core-schema.pan

```diff
@@ -77,18 +77,28 @@ type structure_rule = {
     "table" ? network_valid_routing_table
     @{priority, The priority of the rule over the others. Required by Network Manager when setting routing rules.}
     "priority" ? long(0..0xffffffff)
+    @{nmstate-action used by nmstate module}
+    "nmstate-action" ? choice('blackhole', 'prohibit', 'unreachable')
+    @{nmstate-state used by nmstate module, Can only set to absent for deleting matching route rules}
+    "nmstate-state" ? choice('absent')
+    @{nmstate-iif used by nmstate module, Incoming interface name}
+    "nmstate-iff" ? string
+    @{nmstate-fwmark used by nmstate module. Select the fwmark value to match}
+    "nmstate-fwmark" ? string
+    @{nmstate-fwmask used by nmstate module. Select the fwmask value to match}
+    "nmstate-fwmask" ? string
```

Comment:
Hmm, I'm not sure I like having nmstate specific stuff in the core schema, but right now I can't think of a better way to do it.

Reply:
Acutally, maybe something along the lines of:
```pan
include format('%s-schema', value('/software/components/network/ncm-module', 'basic'));
```

With `nmstate-schema.pan` and `basic-schema.pan` both defining their own version of `structure_rule`?

Reply:
if options are clearly stated is that such a big concern but happy to change it.

however, not sure I fully understand it, so that include goes in core-schema?
I tried separating it into own file then include is in corea-schema,  but tests seem to fail with 
`
Creating man page for target/lib/perl/NCM/Component/nmstate.pm, NCM::Component::nmstate.8[INFO]
[INFO] --- exec-maven-plugin:1.4.0:exec (prove) @ network ---
# VERBOSE No TQU cfg file tqu.cfg found
# VERBOSE No main::TQU cfg string found
# VERBOSE Guessed modules to load: NCM::Component::network
# VERBOSE Namespace regex pattern (?^:^(declaration|unique|object|structure)\stemplate\s(\S+);$)
# VERBOSE Found invalid template components/network/basic-schema.pan (expectedname components/network/basic-schema)
# VERBOSE Found matching template components/network/config-nmstate type unique
# VERBOSE Found matching template components/network/config type unique
# VERBOSE Found matching template components/network/core-schema type declaration
# VERBOSE Found invalid template components/network/nmstate-schema.pan (expectedname components/network/nmstate-schema)
# VERBOSE Found matching template components/network/schema type declaration
#   Failed test 'No invalid pan files found in target/pan (no namespace checked for annotations)'
#   at /v/global/user/k/ka/karimabd/github/configuration-modules-core/ncm-network/target/dependency/build-scripts/Test/Quattor/Doc.pm line 204.
#          got: '2'
#     expected: '0'
`
the structure is fine from what I can see.
happy push it as second commit if you want to see it?

and when I tried testing with a host profile with nmstate backend enabled its seems the path isn't available at that stage, so always includes basic-schema.  Clearly I'm probably including at wrong place, so any guidance here will be appreciated.  

to test, tried this, 
`include format('%s-schema', value('/software/components/network/ncm-module')); `
which results in 
`(/software/components/network/ncm-module) doesn't exist`

Reply:
The first error just looks like the name in the template didn't match the file name `components/network/nmstate-schema.pan`.

The second error is because the `value()` call has no default fall-back value.

Reply:
HI James, 
that's the thing, the name does match filename. I've doubled checked but still throws these errors, it seems any few files I add throws this error. 

yes, the second error is what I've done to see if there is any value of ncm-module even when I set it nmstate, but its still empty. not matter where I include it I can't seem get it to the value so as you say don't think this will work.

the additional nmsate-*  options I added was a good to have, so I can remove them for now if you want to  think about how we would do this, so the other changes can be merged? happy to take on any other suggestions.

Reply:
:disappointed: yeah can you remove them for now then and stick them in another PR? The rest of this looks good to go.

Reply:
I've removed the nmstate-* options for rules, will create a separate pr for that. 
we do need the action options at some point., so will have to find a solution, perhaps we can discuss on next call. thanks.